containers/pixi/mnist_gpu.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# detailed logging to stderr
+set -x
+
+echo -e "# Hello CHTC from Job ${1} running on $(hostname)\n"
+echo -e "# GPUs assigned: ${CUDA_VISIBLE_DEVICES}\n"
+
+echo -e "# Activate Pixi environment\n"
+# The last line of the entrypoint.sh file is 'exec "$@"'. If this shell script
+# receives arguments, exec will interpret them as arguments to it, which is not
+# intended. To avoid this, strip the last line of entrypoint.sh and source that
+# instead.
+. <(sed '$d' /app/entrypoint.sh)
+
+echo -e "\n# Check to see if the NVIDIA drivers can correctly detect the GPU:\n"
+nvidia-smi
+
+echo -e "\n# Extract the training data:\n"
+if [ -f "MNIST_data.tar.gz" ]; then
+    tar -vxzf MNIST_data.tar.gz
+else
+    echo "The training data archive, MNIST_data.tar.gz, is not found."
+    echo "Please transfer it to the worker node in the HTCondor jobs submission file."
+    exit 1
+fi
+
+echo -e "\n# Train a PyTorch CNN classifier on the MNIST dataset:\n"
+# As main.py is copied to the worker through transfer_input_files, it is not
+# located in the same directory as the Pixi manifest (/app/pixi.toml).
+# To avoid moving files around further or complicating the HTCondor submit
+# description file have the command written out here instead of using a Pixi
+# task.
+time pixi run python ./main.py --epochs 20 --save-model
```

Comment:
This is just explaining why in the `pixi/` example we can use the Pixi task runner system

https://github.com/CHTC/templates-GPUs/blob/8db14b24384af4fa3d885cb105e5b2758f697aef/pixi/mnist_gpu.sh#L33

but here we give the command explicitly.

Reply:
Note to self to edit: We don't need the run command, as the environment is already activated.